examples/SMAC4HPO_hyperband_mlp_instances.py

```diff
@@ -0,0 +1,156 @@
+"""
+An example for the usage of SMAC within Python.
+We optimize a simple MLP on the MNIST digits dataset using "Hyperband" intensification.
+
+In this example, we use instances as the budget in hyperband and optimize the average cross validation accuracy.
+An "Instance" represents a specific scenario/condition (eg: different datasets, subsets, transformations)
+for the algorithm to run. SMAC then returns the algorithm that had the best performance across all the instances.
+In this case, an instance is the number of folds used in cross validation.
+"""
+
+import logging
+import warnings
+
+import numpy as np
+from sklearn.model_selection import cross_val_score
+from sklearn.neural_network import MLPClassifier
+from sklearn.datasets import load_digits
+from sklearn.exceptions import ConvergenceWarning
+
+import ConfigSpace as CS
+from smac.configspace import ConfigurationSpace
+from ConfigSpace.hyperparameters import CategoricalHyperparameter, \
+    UniformFloatHyperparameter, UniformIntegerHyperparameter
+
+from smac.scenario.scenario import Scenario
+from smac.facade.smac_hpo_facade import SMAC4HPO
+from smac.intensification.hyperband import Hyperband
+
+digits = load_digits()
+
+
+# Target Algorithm
+def mlp_from_cfg(cfg, seed, instance, budget, **kwargs):
+    """
+        Creates a MLP classifier from sklearn and fits the given data on it.
+        This is the function-call we try to optimize. Chosen values are stored in
+        the configuration (cfg).
+
+        Parameters
+        ----------
+        cfg: Configuration
+            configuration chosen by smac
+        seed: int or RandomState
+            used to initialize the rf's random generator
+        instance: str
+            used to represent the instance to use (number of folds in this case)
+        budget: float
+            used to set max iterations for the MLP
+
+        Returns
+        -------
+        float
+    """
+    # For deactivated parameters, the configuration stores None-values.
+    # This is not accepted by the MLP, so we replace them with placeholder values.
+    lr = cfg['learning_rate'] if cfg['learning_rate'] else 'constant'
+    lr_init = cfg['learning_rate_init'] if cfg['learning_rate_init'] else 0.001
+    batch_size = cfg['batch_size'] if cfg['batch_size'] else 200
+
+    with warnings.catch_warnings():
+        warnings.filterwarnings('ignore', category=ConvergenceWarning)
+
+        mlp = MLPClassifier(
+            hidden_layer_sizes=[cfg["n_neurons"]] * cfg["n_layer"],
+            solver=cfg['solver'],
+            batch_size=batch_size,
+            activation=cfg['activation'],
+            learning_rate=lr,
+            learning_rate_init=lr_init,
+            max_iter=int(np.ceil(budget)),
+            random_state=seed)
+
+        # returns the cross validation accuracy
+        score = cross_val_score(mlp, digits.data, digits.target, cv=int(instance))
```

Comment:
Isn't the `cross_val_score` non-deterministic if you don't pass a CV object with a fixed random state? If yes, could you please change that (also for the other example)?

Reply:
Thanks for the update, but I think you forgot to change it for the other example.